Softeq.XToolkit.Common.iOS/Extensions/UIViewExtensions.cs

```diff
@@ -25,6 +25,87 @@ public static UIView WithBorder(this UIView view, nfloat borderWidth, CGColor? b
             return view;
         }
 
+        /// <summary>
+        ///     UIView with the colored border for target edges.
+        /// </summary>
+        /// <returns>UIView with border.</returns>
+        /// <param name="view">Target view.</param>
+        /// <param name="borderWidth">Border width.</param>
+        /// <param name="borderColor">Border color.</param>
+        /// <param name="edge">Border edge.</param>
+        public static UIView WithBorder(this UIView view, nfloat borderWidth, CGColor borderColor, UIRectEdge edge)
+        {
+            if (edge == UIRectEdge.None)
+            {
+                return view;
+            }
+
+            if (edge.HasFlag(UIRectEdge.All))
+            {
+                return WithBorder(view, borderWidth, borderColor);
+            }
+
+            if (edge.HasFlag(UIRectEdge.Top))
+            {
+                AddBorder(view, borderWidth, borderColor, UIRectEdge.Top);
+            }
+
+            if (edge.HasFlag(UIRectEdge.Bottom))
+            {
+                AddBorder(view, borderWidth, borderColor, UIRectEdge.Bottom);
+            }
+
+            if (edge.HasFlag(UIRectEdge.Left))
+            {
+                AddBorder(view, borderWidth, borderColor, UIRectEdge.Left);
+            }
+
+            if (edge.HasFlag(UIRectEdge.Right))
+            {
+                AddBorder(view, borderWidth, borderColor, UIRectEdge.Right);
+            }
+
+            return view;
+        }
+
+        private static UIView AddBorder(UIView view, nfloat borderWidth, CGColor borderColor, UIRectEdge edge)
+        {
+#pragma warning disable CS8509 // The switch expression does not handle all possible values of its input type (it is not exhaustive).
+            var autoresizingMask = edge switch
```

Comment:
Maybe we could pass `autoresizingMask` and `frame` from outside instead of switching, as we should already know what type of border we are trying to add? For instance
```
            if (edge.HasFlag(UIRectEdge.Right))
            {
                AddBorder(
                    view, 
                    borderWidth, 
                    borderColor, 
                    UIViewAutoresizing.FlexibleHeight | UIViewAutoresizing.FlexibleLeftMargin,
                    new CGRect(view.Frame.Size.Width - borderWidth, 0, borderWidth, view.Frame.Size.Height));
            }
```
or enumerate through flags in WithBorder(..., UIRectEdge edge) method like this:
```
            foreach (var flag in edge.GetFlags())
            {
                AddBorder(view, borderWidth, borderColor, flag);
            }
            
            where
            
            public static IEnumerable<Enum> GetFlags(this Enum e)
            {
                return Enum.GetValues(e.GetType()).Cast<Enum>().Where(e.HasFlag);
            }
```

Reply:
Passing `autoresizingMask` and `frame` good idea
Done.

foreach can't be simply applied because we need different ways for some elements